### PR TITLE
config: remove placeholder secret

### PR DESCRIPTION
**Pull request by cirocosta**
## Changes proposed by this PR

- remove the [placeholder secret][carvel-placeholder-secrets] that was in place aimed at the carvel packaging of Cartographer in favor of doing so in [package-for-cartographer's image-pull-secret overlay](https://github.com/vmware-tanzu/package-for-cartographer/blob/f2f2f1b6c93c0a719f5a401275c875ecd346c73e/src/cartographer/config/overlays/image-pull-secret.yaml)

[carvel-placeholder-secrets]: https://github.com/vmware-tanzu/carvel-secretgen-controller/blob/63766ca8cbc1270fd58312c8cea427f4dae9a181/docs/secret-export.md#placeholder-secrets

in order to make the installation of Cartographer in carvel-based
packaging easier, we used to have this secret that gets filled by
secretgen-controller as part of the set of objects shipped in the
release, but given that it's such a packaging-specific detail, here we
get rid of it favoring `package-for-cartographer` as the location for
doing so.

see https://github.com/vmware-tanzu/package-for-cartographer/pull/23 for
the PR where that addition of a secret to the package takes place (file:
src/cartographer/config/overlays/image-pull-secret.yaml)

## Release Note

- Remove secretgen-controller placeholder secret from Cartographer release

## PR Checklist

Note: Please do not remove items. Mark items as done `[x]` or use ~strikethrough~ if you believe they are not relevant

- [ ] Linked to a relevant issue. Eg: `Fixes #123` or `Updates #123`
- [ ] Removed non-atomic or `wip` commits
- [ ] Filled in the [Release Note](#Release-Note) section above 
- [ ] Modified the docs to match changes

### CODE REVIEW

**Comment by cirocosta** [comment]
~uhhh `hack/setup.sh` when building the release that we use in the tests actually pushes to the local registry that is enforcing authentication, so e2e tests will currently fail. marking it draft until there's a fix in place.~

done

File: hack/setup.sh [hunk 1]
```diff
@@ -176,30 +173,13 @@ display_vars() {
 }
 
 start_registry() {
-        log "starting registry"
```
Comment (scothis) [inline hack/setup.sh:179]: was removing the log line intentional?

File: packaging/overlays/strip-status.yaml [hunk 2]
```diff
@@ -1,20 +0,0 @@
-# Copyright 2021 VMware
```
Comment (scothis) [inline packaging/overlays/strip-status.yaml:1]: don't we still need this?

Comment (scothis) [reply packaging/overlays/strip-status.yaml:1]: ah, I see it's in the package repo https://github.com/vmware-tanzu/package-for-cartographer/blob/main/src/cartographer/config/overlays/strip-status.yaml

File: hack/setup.sh (hunk 1 above)
Comment (cirocosta) [reply hack/setup.sh:179]: whopssy

**Review by martyspiewak — APPROVED** [review]

**Review by scothis — APPROVED** [review]

**Pull request merged**